botocore/credentials.py

```diff
@@ -1415,8 +1449,25 @@ def _resolve_credentials_from_profile(self, profile_name):
         profiles = self._loaded_config.get('profiles', {})
         profile = profiles[profile_name]
 
-        if self._has_static_credentials(profile):
+        if self._has_static_credentials(profile) and \
+                not self._profile_provider_builder:
+            # This is only here for backwards compatibility
```

Comment:
I believe the intent was that in case this was being instantiated without the profile provider builder, it should still function for the static credential / assume role case that it did before this change.

Reply:
Can you add that to the comment?

Reply:
Sorry didn't see this, updated the commit as a revision to my last commit and force pushed.